network/src/peers.rs

```diff
@@ -560,6 +600,12 @@ impl<N: Network, E: Environment> Peers<N, E> {
                 if let Err(error) = outbound.send(message).await {
                     trace!("Outbound channel failed: {}", error);
                     self.connected_peers.write().await.remove(&peer);
+
+                    #[cfg(any(feature = "test", feature = "prometheus"))]
+                    {
+                        let number_of_connected_peers = self.number_of_connected_peers().await;
+                        metrics::gauge!(metrics::peers::CONNECTED, number_of_connected_peers as f64);
+                    }
```

Comment:
this metric would ideally only be updated at `PeersRequest::{PeerConnected, PeerDisconnected}`; would it be possible to instead remove `self.connected_peers.write().await.remove(&peer);`, the selected section, and instead to send a `PeerDisconnected` message to the peer handler from here?

Reply:
I agree, though I think this implies changes to logic outside of metrics, happy to submit a follow up to address this though 👍